connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/adhoctoolsschema/resolver/CachingAdHocToolsSchemaResolver.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.agenticai.adhoctoolsschema.resolver;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import io.camunda.connector.agenticai.adhoctoolsschema.model.AdHocToolsSchemaResponse;
+import java.time.Duration;
+import java.util.Optional;
+
+public class CachingAdHocToolsSchemaResolver implements AdHocToolsSchemaResolver {
```

Comment:
Simple caching decorator holding resolve tool definition in memory - size and retention time is configurable and uses very small defaults, so I think we're not risking filling up memory.